### PR TITLE
#168362228 setting up database and connect Endpoints

### PR DESCRIPTION
**Pull request by victor-abz**
#### What does this PR do?
Setting server
using DB
> User can signup 
> User can log in
> Admin can change a User to Mentor -- using DB
> Can get all the Users

#### Description of Task to be completed
connected our app with Postgres,
Set up a server with database
connected user-related endpoints to the Database

#### How should this be manually tested?
Clone this branch and run
```
npm install
```
then using postman, can test 
```
api/v1/auth/signup -- for sign up
api/v1/auth/login -- for login
API/v1/user/:userId -- for change a user to mentor
/api/v1/users -- to get all users
```
#### What are the relevant pivotal tracker stories?
[finish #168362228][start #168362249][start #168362248]

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import http from 'http';
 import express from 'express';
 import logger from 'morgan';
-import bodyParser from 'body-parser'; // parsing bodies
+import dotenv from 'dotenv';
+import bodyParser from 'body-parser';
 import swaggerUi from 'swagger-ui-express';
+// import db from './server/db'
+
```
Comment (njeri-ngigi) [inline app.js]: Please remove commented out code. It can make you confused as you continue

File: server/controllers/sessionController.js [hunk 2]
```diff
@@ -1,120 +1,60 @@
 import sessionModel from '../models/sessions';
 import reviewModel from '../models/review';
+import Helper from '../helpers/helper';
 
 const sessionController = {
-// Creating Session
+  // Creating Session
```
Comment (njeri-ngigi) [inline server/controllers/sessionController.js]: no need to unnecessarily comment. Using descriptive variable names is a more effective way of documenting your code than comments.

Only use comments for code that you're sure needs explaining or is slightly confusing or if altering that code could cause side-effects

File: server/controllers/sessionController.js [hunk 3]
```diff
@@ -1,120 +1,60 @@
...
-      return res.status(200).json({
-        status: 200,
-        data,
-      });
+    const roles = ['user', 'mentor', 'admin'];
+    const userRole = Helper.getRole(req.userData.role);
+
+    if (roles.indexOf(req.userData.role) === -1) {
+      const status = 401;
+      const error = 'Insufficient provilege. Please sign in';
```
Comment (njeri-ngigi) [inline server/controllers/sessionController.js]: Typo `privilege` instead of `provilege`

File: server/db/index.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      lastName, 
+      email, 
+      password, 
+      address, 
+      bio, 
+      occupation, 
+      expertise, 
+      role) 
+    values(
+      '${data.firstName}',
```
Comment (njeri-ngigi) [inline server/db/index.js]: make use of object destructuring. E.g.

```
const { lastName, email, hashPassword, address, bio, occupation, expertise } = data;

File: server/db/index.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    const userFound = await this.findByProp('users', 'email', data.email);
+    const userExist = userFound.rows[0];
+
+    if (!Helper.comparePassword(userExist.password, data.password)) {
+      return Helper.handleError(res, 400, 'The credentials you provided is incorrect');
+    }
+
+    const payload = {
+      email: userExist.email,
+      userId: userExist.userId,
```
Comment (njeri-ngigi) [inline server/db/index.js]: make use of object destructuring here

File: server/db/index.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    const result = await con.query(`SELECT COUNT(1) FROM ${table} WHERE ${column} = '${value}';`);
+    return result;
+  }
+  // async addSession(data) {
+  // 	const newSession = await con.query(`Insert into sessions
+  //		 (mentorId, menteeId, questions, menteeEmail, status) values(
+  // 		'${data.mentorId}',
+  // 		'${data.menteeId}',
+  // 		'${data.questions}',
+  // 		'${data.menteeeEmail}',
```
Comment (njeri-ngigi) [inline server/db/index.js]: remove commented out code

File: server/db/index.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+            CREATE TABLE IF NOT EXISTS REVIEWS (sessionId INTEGER REFERENCES sessions(sessionId) ON DELETE CASCADE,menteeId INTEGER REFERENCES users(userId) ON DELETE CASCADE,mentorId INTEGER REFERENCES users(userId) ON DELETE CASCADE,menteefirstName VARCHAR(300),menteelastName VARCHAR(300),remark VARCHAR(300));
+        `);
+
+    const result = await this.findUsers('users', 'email', 'admin@freementors.com');
+    if (result.rows[0].count === '0') {
+      const payload = {
+        firstName: 'admin',
+        lastName: 'test',
+        email: 'admin@freementors.com',
+        password: 'mentor',
```
Comment (njeri-ngigi) [inline server/db/index.js]: exposing your admin password like this makes your codebase vulnerable. Place this in the `.env` file

File: app.js (hunk 1 above)
Comment (victor-abz) [reply app.js]: OK

**Pull request closed without merging**